libp2p/Host.cpp

```diff
@@ -328,21 +289,17 @@ void Host::runAcceptor()
 	{
 		clog(NetConnect) << "Listening on local port " << m_listenPort << " (public: " << m_tcpPublic << ")";
 		m_accepting = true;
-		m_socket.reset(new bi::tcp::socket(m_ioService));
-		m_tcp4Acceptor.async_accept(*m_socket, [=](boost::system::error_code ec)
+		
+		bi::tcp::socket* s = new bi::tcp::socket(m_ioService);
+		m_tcp4Acceptor.async_accept(*s, [=](boost::system::error_code ec)
```

Comment:
I do not understand you response. If the client is accepting, the lambda is not called? In that case, s is not deallocated. PLEASE use modern memory management concepts, especially if there is only a tiny grain of uncertainty about the lifetime of objects or the call structure of the code.